### ISSUE TITLE
Incorrect pixel offset between marks and axes

### ISSUE THREAD

**Issue by kanitw** [vega/vega#845]
In order to support `size.value` for band scale in VL, I'm using `xc` with `offset = band: 0.5` but some how I need an extra 1 pixel offset to make the bar appear centered.  Is this an intended behavior?

```
              "xc": {
                "scale": "x",
                "field": "age",
                "offset": {
                  "scale": "x",
                  "band": 0.5,
                  "offset": 1       // Somehow I need to add this 1 pixel extra offset 
                }
              },
```

<img width="394" alt="vega_editor" src="https://cloud.githubusercontent.com/assets/111269/26815618/2833aab6-4a42-11e7-8bde-093801313cb4.png">

Full spec here: https://pastebin.com/8pqkP8ia

(Try changing `width.value` to `1` makes it even clearer that we need an extra 1 pixel offset.)

**Comment by jheer** [comment]
Thanks! I looked into this and the problem is not with the marks, but with the axes. It turns out that the `0.5` pixel offset we use to ensure "crisp" rendering causes the misalignment. Fortunately, it seems that changing this to a `-0.5` pixel offset causes everything to align correctly. I'll fix this later, at which point you can remove the 1 pixel offset.

This may also relate to vega/vega-scenegraph#41 cc: @domoritz

Finally, your encoding is unnecessarily complicated. The following is equivalent:
```json
"xc": {
  "scale": "x",
  "field": "age",
  "band": 0.5,
  "offset": 1
},
```

**Comment by jheer** [comment]
After further investigation and testing, this turns out to be more insidious than I originally thought. Changing the axis offset from `0.5` to `-0.5` fixes the tick centering issue, but introduces new (arguably more problematic) misalignment issues elsewhere. For example, it can cause new boundary misalignments at the beginning and end of the chart for linear scales.

An alternative fix is to automatically offset the `xc` or `yc` calculation by a pixel, but that also causes problems elsewhere, as it actually de-centers the mark relative to the true band size.

The heart of the issue is the use of single pixel strokes that are aligned to the pixel grid, even though the strokes are (mathematically speaking) more accurately placed in-between pixels. Of course, that results in aliasing that we do not want. I looked at the output of some other tools and noted that by default ggplot2 uses 2 pixels for major ticks and gridlines to address this issue: this places the tick in the "correct" place and the 2 pixel width avoids aliasing. But I don't think we want to require such "fat" ticks either.

So, it seems that for your case of centered ticks an offset is needed. To address the issue I plan to add a new `tickOffset` config parameter for axes, and update the default Vega config to include a `-1` offset for band and point scales. This means you won't need to do anything at the Vega-Lite level (just remove your custom offset) so long as you want centered ticks for band/point scales. If you instead want to use boundary ticks (e.g., to show an interval, as done in the current violin-plot example), the config will need to be adjusted to `"tickOffset": 0`. This is not an elegant solution, but so far is the best solution I've found in terms of trade-offs.

Let me know if you have any thoughts or feedback. I'll comment here again once such changes are tested, committed, and published.
